src/sentry/api/endpoints/dif_files.py

```diff
@@ -6,14 +6,28 @@
 from rest_framework.response import Response
 
 from sentry.utils import json
+from sentry.api.serializers import serialize
 from sentry.api.bases.chunk import ChunkAssembleMixin
 from sentry.api.bases.project import ProjectEndpoint, ProjectReleasePermission
-from sentry.models import File, ChunkFileState
+from sentry.models import File, ChunkFileState, ProjectDSymFile
 
 
 class DifAssembleEndpoint(ChunkAssembleMixin, ProjectEndpoint):
     permission_classes = (ProjectReleasePermission, )
 
+    def _add_project_dsym_to_reponse(self, found_file, response):
+        if found_file is not None:
+            if found_file.headers.get('error', None) is not None:
+                response['error'] = found_file.headers.get('error')
```

Comment:
Maybe just a minor consistency thing, but: What if there is an `error` header but the state is `OK`? In that case it is weird, that you're adding that field.

Reply:
The database doesn't prevent this from happening, but there is no codepath setting `OK` and `error`. So when there is an `error` state is also an `error`.